rbdl/src/Kinematics.cc

```diff
@@ -378,6 +378,39 @@ RBDL_DLLAPI void CalcPointJacobian6D (
   }
 }
 
+RBDL_DLLAPI void calcLinkDependency (
+    Model &model,
+    unsigned int body_id,
+    MatrixNd &J) 
+{      
+  std::vector<int> active_joints;
+  unsigned int reference_body_id = body_id;
+
+  if (model.IsFixedBodyId(body_id)) {
+    unsigned int fbody_id   = body_id
+      - model.fixed_body_discriminator;
+
+    reference_body_id       = model
+      .mFixedBodies[fbody_id]
+      .mMovableParent;
+  } 
+
+  unsigned int j = reference_body_id;
+
+  while (j != 0) {
+    unsigned int q_index = model.mJoints[j].q_index;
+    active_joints.push_back(q_index);
+    j = model.lambda[j];
+  }
+
+  std::sort(active_joints.begin(), active_joints.end());
+  J = MatrixNd::Zero(active_joints.size(), model.q_size);
+  for (int i = 0; i < active_joints.size(); ++i) {
+    J(i, active_joints[i]) = 1;
```

Comment:
This does not take into account robot models with spherical joints between the base and the required body id. Either add this case, or throw an error in the case a spherical joint is encountered

Reply:
Added error case for spherical joint